### PR TITLE
Create restaurants.py

### PR DESCRIPTION
**Pull request by BreckoD**
Created a DataProvider for NYC Restaurant Inspection data.

### CODE REVIEW

File: src/sttn/data/restaurants.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+from sttn import network
+from .data_provider import DataProvider
+
+class RestaurantInspectionDataProvider(DataProvider):
+    """New York City Restaurant Inspections data provider, builds a network where nodes represent restaurants
+     and every edge represents an inspection where the origin and destination point to the node
+     where the inspection happened."""
+  
+    @staticmethod
+    def get_data(self, from_date, to_date) -> network.SpatioTemporalNetwork:
```
Comment (yuribogomolov) [inline src/sttn/data/restaurants.py]: It's good practice to add the typing information for function arguments: `from_date: str`

File: src/sttn/data/restaurants.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        Args:
+            from_date (str): A string with year and month and in the "YYYY-MM-DD" format.
+            Specifies the start date to get data from
+            
+            to_date (str): A string with year, month, and day in the "YYYY-MM-DD" format.
+            Specifies the end date to get data from
+                
+
+        Returns:
```
Comment (yuribogomolov) [inline src/sttn/data/restaurants.py]: Please add column-level documentation (see [this example](https://github.com/yuribogomolov/sttn/blob/main/src/sttn/data/nyc.py#L44))

**Review by yuribogomolov — APPROVED** [review]
Looks good to me! Please address the two comments below before merging

**Comment by BreckoD** [comment]
Addressed your comments in the last commit. Next time, I will add the new version with a different network structure to the nyc.py. So, the restaurants.py can be used while I'm working on the updated version.

**Comment by BreckoD** [comment]
Now it should be 100% ready

**Pull request merged**